src/main/java/com/oltpbenchmark/benchmarks/templated/procedures/GenericQuery.java

```diff
@@ -36,11 +36,32 @@ public abstract class GenericQuery extends Procedure {
 
     /** Execution method with parameters. */
     public void run(Connection conn, List<Object> params) throws SQLException {
-        try (PreparedStatement stmt = getStatement(conn, params); ResultSet rs = stmt.executeQuery()) {
-            while (rs.next()) {
-                //do nothing
+
+        try (PreparedStatement stmt = getStatement(conn, params)) {
+
+            // False for UPDATE, INSERT, DELETE queries
+            boolean hasResultSet = stmt.execute();
+
+            while (true) {
+                if (hasResultSet) {
+                    try (ResultSet rs = stmt.getResultSet()) {
+                        while (rs.next()) {
+                            // do nothing
+                        }
+                    }
+                } else {
+                    if (stmt.getUpdateCount() == -1) {
```

Comment:
Why do you need this additional check?

Reply:
From Brian's pinned Stackoverflow post:
_"The loop is there because some database systems (e.g. SQL Server), can produce multiple result sets and multiple update counts, and this will handle all of them."_